sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeBatchActionsAsyncClient.java

```diff
@@ -311,45 +319,71 @@ private AnalyzeBatchActionsResult toAnalyzeTasks(AnalyzeJobState analyzeJobState
-            extractKeyPhrasesActionResults = IterableStream.of(keyPhraseExtractionTasks.stream()
-                .map(taskItem -> {
-                    ExtractKeyPhrasesActionResult actionResult = new ExtractKeyPhrasesActionResult();
-                    ExtractKeyPhrasesActionResultPropertiesHelper.setResult(actionResult,
-                        toExtractKeyPhrasesResultCollection(taskItem.getResults()));
-                    TextAnalyticsActionResultPropertiesHelper.setCompletedAt(actionResult,
-                        taskItem.getLastUpdateDateTime());
-                    return actionResult;
-                })
-                .collect(Collectors.toList()));
+            for (int i = 0; i < keyPhraseExtractionTasks.size(); i++) {
+                final TasksStateTasksKeyPhraseExtractionTasksItem taskItem = keyPhraseExtractionTasks.get(i);
+                final ExtractKeyPhrasesActionResult actionResult = new ExtractKeyPhrasesActionResult();
+                ExtractKeyPhrasesActionResultPropertiesHelper.setResult(actionResult,
+                    toExtractKeyPhrasesResultCollection(taskItem.getResults()));
+                TextAnalyticsActionResultPropertiesHelper.setCompletedAt(actionResult,
+                    taskItem.getLastUpdateDateTime());
+                extractKeyPhrasesActionResults.add(actionResult);
+            }
+        }
+
+        final List<TextAnalyticsError> errors = analyzeJobState.getErrors();
+        if (!CoreUtils.isNullOrEmpty(errors)) {
+            for (TextAnalyticsError error : errors) {
+                final String[] targetPair = parseActionErrorTarget(error.getTarget());
+                final String taskName = targetPair[0];
+                final Integer taskIndex = Integer.valueOf(targetPair[1]);
+                final TextAnalyticsActionResult actionResult;
+                if ("entityRecognitionTasks".equals(taskName)) {
+                    actionResult = recognizeEntitiesActionResults.get(taskIndex);
+                } else if ("entityRecognitionPiiTasks".equals(taskName)) {
+                    actionResult = recognizePiiEntitiesActionResults.get(taskIndex);
+                } else if ("keyPhraseExtractionTasks".equals(taskName)) {
+                    actionResult = extractKeyPhrasesActionResults.get(taskIndex);
+                } else {
+                    throw logger.logExceptionAsError(new RuntimeException(
+                        "Invalid task name in target reference, " + taskName));
+                }
+
+                TextAnalyticsActionResultPropertiesHelper.setIsError(actionResult, true);
+                TextAnalyticsActionResultPropertiesHelper.setError(actionResult,
+                    new com.azure.ai.textanalytics.models.TextAnalyticsError(
+                        TextAnalyticsErrorCode.fromString(
+                            error.getCode() == null ? null : error.getCode().toString()),
+                        error.getMessage(), error.getTarget()));
```

Comment:
consider not adding the target to the error as this won't provide useful information for the user.